### DISCUSSION TITLE
JSON dictionary format

### DISCUSSION THREAD

**Discussion by aeklo** [strandborg/AlsavoCtrl#3] [Ideas]
Hi, and thanks for taking the time to write this command :-)

I use Home Assistant for automation, and would like to import the AlsavoCtrl values. Home Assistant has support for JSON dictionaries, but the output from AlsavoCtrl is not a dictionary. Any chance this will be a feature?

**Comment by strandborg** [comment]
Hi, the source code should be super easy to modify for your purposes, the JSON printout happens in AlsavoCtrl.cpp lines 622 and 625. It is currently just thrown together so that my existing MQTT scripts (with JSON transformers) could pick them up.

**Reply by aeklo** [reply]
Thanks, I'll take a look :-)

**Reply by aeklo** [reply]
Worked perfectly! Thanks!

**Reply by rabbadab** [reply]
Hey! @aeklo will you make this into a Home assistant integration? That would be so great. Have been looking for this for several years!

**Reply by aeklo** [reply]
Unfortunately I neither have the time or knowledge to make a complete integration myself, but head over to the HA forum discussion. Maybe we can get some collaboration going for an integration :)
[https://community.home-assistant.io/t/alsavo-pro-heat-pump/308072](url)

**Reply by gllmlbrt** [reply]
https://github.com/gllmlbrt/Alsavo-HassOS
I resorted to use NodeRed essentially to do the string to JSON. Very inelegant. but working so far.
